storage/range_tree.go

```diff
@@ -443,3 +454,407 @@ func (tc *treeContext) insertCase5(node *proto.RangeTreeNode) error {
 	}
 	return nil
 }
+
+// DeleteRange removes a range from the RangeTree. This should only be called
+// from operations that remove ranges, such as AdminMerge.
+func DeleteRange(txn *client.Txn, b *client.Batch, key proto.Key) error {
+	tc, err := getRangeTree(txn)
+	if err != nil {
+		return err
+	}
+	node, err := tc.getNode(key)
+	if err != nil {
+		return err
+	}
+	if node == nil {
+		return util.Errorf("could not find range tree node with the key %s", key)
+	}
+	err = tc.delete(node)
```

Comment:
the TL;DR is that it's preferable to shadow in all cases where the shadowing is unproblematic. that is, if you can avoid ever writing `err =` for a particular scope, you should. If you can't avoid that assignment then you need to be more deliberate about shadowing vs assignment. this is probably easier to describe deeply in person.